src/parser/github-comment-module.ts

```diff
@@ -0,0 +1,212 @@
+import { Value } from "@sinclair/typebox/value";
+import Decimal from "decimal.js";
+import * as fs from "fs";
+import { stringify } from "yaml";
+import configuration from "../configuration/config-reader";
+import githubCommentConfig, { GithubCommentConfiguration } from "../configuration/github-comment-config";
+import { getOctokitInstance } from "../get-authentication-token";
+import { CommentType, IssueActivity } from "../issue-activity";
+import { parseGitHubUrl } from "../start";
+import { getPayoutConfigByNetworkId } from "../types/payout";
+import program from "./command-line";
+import { GithubCommentScore, Module, Result } from "./processor";
+
+/**
+ * Posts a GitHub comment according to the given results.
+ */
+export class GithubCommentModule implements Module {
+  private readonly _configuration: GithubCommentConfiguration = configuration.githubComment;
+  private readonly _debugFilePath = "./output.html";
+
+  async transform(data: Readonly<IssueActivity>, result: Result): Promise<Result> {
+    let body = "";
```

Comment:
I've always been anti empty string for variable declarations. It's let to too many problems with other less experienced developers. I wish for a linter rule to flag these as a consequence. 

Consider initializing an empty array as a buffer and then joining it at the end.

Reply:
I think in this case it makes sense because this function will return `""` if nothing is found. If I return `undefined` or `null` it will be interpreted in HTML as a string and will display the text null or undefined.

I changed the code to always return `content` so it forces it to be properly initialized which should avoid mistakes.

Reply:
[].join(""); yields that

Reply:
Seems convoluted to me and I don't know what problem that solved, but sure can do. Because either way this will yield an empty string or populated string?

Reply:
It might seem convoluted but I think that it condenses the logic to coerce everything into a string as expected. This is a suggestion to be proactive when we set up the empty string linter one day. 

For example if you `push` any element into the array (compared to `+=`) when you join they should render as expected.  

If you do `"1" += 1` you will get a different result

Reply:
In the case of a string it would coerce the value and you would get "null" as a string so I get your point indeed

Reply:
Will you use array syntax?

Reply:
Had changed it everywhere else but here, now it should be fine